internal/request.go

```diff
@@ -27,6 +27,27 @@ func (s *SPOA) processRequest(msg spoe.Message) ([]spoe.Action, error) {
 		tx      types.Transaction
 	)
 	var app *application
+
+	defer func() {
+		if tx == nil || tx.ID() == "" || app == nil {
```

Comment:
I wonder if the case `tx.ID() == ""` is needed.

Reply:
Looks like [not needed](https://github.com/corazawaf/coraza/blob/ad50864cfa00f5143fff7f6877847326fdc4b394/internal/corazawaf/waf.go#L140)